Assets/HoloToolkit/UX/Scripts/Lines/LineStripMesh.cs

```diff
@@ -0,0 +1,161 @@
+        private List<Color> colors = new List<Color>();
+        private List<float> widths = new List<float>();
+
+        private GameObject meshRendererGameObject;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+
+            if (LineMaterial == null)
+            {
+                Debug.LogError("Line material cannot be null.");
+                enabled = false;
+                return;
+            }
+
+            lineMatInstance = new Material(LineMaterial);
+
+            // Create a mesh
+            if (stripMesh == null)
+            {
+                stripMesh = new Mesh();
+            }
+
+            if (stripMeshRenderer == null)
+            {
+                meshRendererGameObject = new GameObject("Strip Mesh Renderer");
+                stripMeshRenderer = meshRendererGameObject.AddComponent<MeshRenderer>();
+            }
+            stripMeshRenderer.sharedMaterial = lineMatInstance;
+            stripMeshRenderer.shadowCastingMode = UnityEngine.Rendering.ShadowCastingMode.Off;
+            stripMeshRenderer.receiveShadows = false;
+            stripMeshRenderer.lightProbeUsage = UnityEngine.Rendering.LightProbeUsage.Off;
+
+            MeshFilter stripMeshFilter = stripMeshRenderer.GetComponent<MeshFilter>();
+            if (stripMeshFilter == null)
+                stripMeshFilter = stripMeshRenderer.gameObject.AddComponent<MeshFilter>();
+
+            stripMeshFilter.sharedMesh = stripMesh;
+        }
+
+        private void OnDisable()
+        {
+            if (lineMatInstance != null)
+                GameObject.Destroy(lineMatInstance);
+
+            if (meshRendererGameObject != null)
+            {
+                GameObject.Destroy(meshRendererGameObject);
+                stripMeshRenderer = null;
+            }
+        }
+
+        public void Update()
+        {
+            if (!source.enabled)
+            {
+                stripMeshRenderer.enabled = false;
+                return;
+            }
+
+            stripMeshRenderer.enabled = true;
+            positions.Clear();
+            forwards.Clear();
+            colors.Clear();
+            widths.Clear();
+
+            for (int i = 0; i <= NumLineSteps; i++)
+            {
+                float normalizedDistance = (1f / (NumLineSteps - 1)) * i;
+                positions.Add(source.GetPoint(normalizedDistance));
+                colors.Add(GetColor(normalizedDistance));
+                widths.Add(GetWidth(normalizedDistance));
+                forwards.Add(source.GetRotation(normalizedDistance) * Vector3.down);
+            }
+
+            GenerateStripMesh(positions, colors, widths, uvOffset, forwards, stripMesh);
+        }
+
+        public static void GenerateStripMesh(List<Vector3> positionList, List<Color> colorList, List<float> thicknessList, float uvOffsetLocal, List<Vector3> forwardList, Mesh mesh)
+        {
+            // TODO store these as local variables to reduce allocations
```

Comment:
Let's go ahead and do this TODO. seems simple enough

Reply:
Done.